From efffe4e6f0aa377386732907c533d6d938b12921 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 24 Aug 2017 17:31:46 +0800
Subject: [PATCH 80/87] dt-bindings: net: dwmac-sun8i: document the binding of
 the R40 GMAC

Allwinner R40 SoC features a GMAC like the MACs in Allwinner
A83T/H3/V3s/A64 SoCs, but with the syscon register moved to the CCU
memory region.

Add the device binding of the R40 GMAC.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 .../devicetree/bindings/net/dwmac-sun8i.txt        | 23 ++++++++++++++++------
 1 file changed, 17 insertions(+), 6 deletions(-)

diff --git a/Documentation/devicetree/bindings/net/dwmac-sun8i.txt b/Documentation/devicetree/bindings/net/dwmac-sun8i.txt
index 725f3b187886..4d83b5e18380 100644
--- a/Documentation/devicetree/bindings/net/dwmac-sun8i.txt
+++ b/Documentation/devicetree/bindings/net/dwmac-sun8i.txt
@@ -7,6 +7,7 @@ Required properties:
 - compatible: should be one of the following string:
 		"allwinner,sun8i-a83t-emac"
 		"allwinner,sun8i-h3-emac"
+		"allwinner,sun8i-r40-gmac"
 		"allwinner,sun8i-v3s-emac"
 		"allwinner,sun50i-a64-emac"
 - reg: address and length of the register for the device.
@@ -20,12 +21,6 @@ Required properties:
 - phy-handle: See ethernet.txt
 - #address-cells: shall be 1
 - #size-cells: shall be 0
-- syscon: A phandle to the syscon of the SoC with one of the following
- compatible string:
-  - allwinner,sun8i-h3-system-controller
-  - allwinner,sun8i-v3s-system-controller
-  - allwinner,sun50i-a64-system-controller
-  - allwinner,sun8i-a83t-system-controller
 
 Optional properties:
 - allwinner,tx-delay-ps: TX clock delay chain value in ps. Range value is 0-700. Default is 0)
@@ -33,6 +28,22 @@ Optional properties:
 Both delay properties need to be a multiple of 100. They control the delay for
 external PHY.
 
+Required properties for the following compatibles:
+  - "allwinner,sun8i-a83t-emac",
+  - "allwinner,sun8i-h3-emac",
+  - "allwinner,sun8i-v3s-emac",
+  - "allwinner,sun50i-a64-emac":
+- syscon: A phandle to the syscon of the SoC with one of the following
+ compatible string:
+  - allwinner,sun8i-a83t-system-controller
+  - allwinner,sun8i-h3-system-controller
+  - allwinner,sun8i-v3s-system-controller
+  - allwinner,sun50i-a64-system-controller
+
+Required properties for the following compatibles:
+  - "allwinner,sun8i-r40-gmac":
+- allwinner,ccu: A phandle to the CCU of the SoC
+
 Optional properties for the following compatibles:
   - "allwinner,sun8i-h3-emac",
   - "allwinner,sun8i-v3s-emac":
-- 
2.13.5

